src/Raven.Server/Documents/Handlers/FullTopologyHandler.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Raven.Abstractions.Data;
+using Raven.Abstractions.Replication;
+using Raven.Client.Replication.Messages;
+using Raven.Server.Documents.Replication;
+using Raven.Server.Json;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class FullTopologyHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/full-topology", "GET")]
+        public async Task GetFullTopology()
+        {
+            DocumentsOperationContext context;
+            using (Database.DocumentsStorage.ContextPool.AllocateOperationContext(out context))
+            {
+                ReplicationDocument replicationDocument;
+                using (context.OpenReadTransaction())
+                {
+                    var configurationDocument = Database.DocumentsStorage.Get(context, Constants.Replication.DocumentReplicationConfiguration);
+                    if (configurationDocument == null)
+                    {
+                        WriteEmptyTopology(context);
+                        return;
+                    }
+
+                    replicationDocument = JsonDeserializationServer.ReplicationDocument(configurationDocument.Data);
+                    if (replicationDocument.Destinations?.Count == 0)
+                    {
+                        WriteEmptyTopology(context);
+                        return;
+                    }
+                }
+
+                using (var clusterTopologyExplorer = new ReplicationClusterTopologyExplorer(
+                    Database,
+                    new Dictionary<string, List<string>>(),
+                    (long)Database.Configuration.Replication.ReplicationTopologyDiscoveryTimeout.AsTimeSpan.TotalMilliseconds,
```

Comment:
you are passing long, which is a part of TimeSpan and inside ReplicationClusterTopologyExplorer you are building TimeSpan from that long again, just pass the TimeSpan?